### PR TITLE
INFRA: Add Application Insight

### PR DESCRIPTION
**Pull request by Pwongchaiya**

### CODE REVIEW

File: InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ApplicationInsight.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace InternTrack.Core.Api.Infrastructure.Provision.Brokers.Clouds
+{
+    public partial class CloudBroker
+    {
+        public async ValueTask<ApplicationInsightsComponentResource> CreateApplicationInsightComponentAsync(
+            string componentName,
+            ResourceGroupResource resourceGroup)
+        {
+            ApplicationInsightsComponentCollection appInsightCollection = resourceGroup.GetApplicationInsightsComponents();
```
Comment (glhays) [inline InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ApplicationInsight.cs:21]:
```suggestion
            ApplicationInsightsComponentCollection appInsightCollection = resourceGroup.GetApplicationInsightsComponents();
```
            ApplicationInsightsComponentCollection appInsightCollection =
```             resourceGroup.GetApplicationInsightsComponents();

File: InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (glhays) [inline InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj:1]: Is this file needed.

File: InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ApplicationInsight.cs (hunk 1 above)
Comment (Pwongchaiya) [reply InternTrack.Core.Api.Infrastructure.Provision/Brokers/Clouds/CloudBroker.ApplicationInsight.cs]: I caught that yesterday too. Could've of sworn it got pushed. But thank you!

File: InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj (hunk 2 above)
Comment (glhays) [reply InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj:1]: InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj

InternTrack - Backup???

Comment (Pwongchaiya) [reply InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj:1]: Sdk="Microsoft.NET.Sdk was needed. I dont know why exactly but provision layer would'nt reload after i deleted it.

Comment (Pwongchaiya) [reply InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj:1]: I don't know where it went. What should i do?

Comment (glhays) [reply InternTrack.Core.Api.Infrastructure.Provision/InternTrack - Backup.Core.Api.Infrastructure.Provision.csproj:1]: Do you see the odd naming of this file? Maybe a git thing.

Yes the line you removed is typically needed in the .csproj file.
